microsoft-azure-storage-test/src/com/microsoft/azure/storage/blob/CloudBlockBlobTests.java

```diff
@@ -1567,6 +1579,101 @@ private void doUploadDownloadFileTest(CloudBlockBlob blob,  byte[] fileBuffer, i
         }
     }
 
+    //Following test case expect "bigFile.mp4" into resource folder of test module
+    @Test
+    @Category({ DevFabricTests.class, DevStoreTests.class, SlowTests.class })
+    public void testBigFileUploadWithSingleThread() throws IOException, StorageException, URISyntaxException {
+
+        InputStream inputStream = getBigFileToUpload();
+        final CloudBlockBlob blob = getCloudBlockBlob("bigFileTestBlob_");
+
+        this.doUploadBigFileTest(blob, inputStream, 4 * Constants.MB, 1, false, false);
+    }
+
+    //Following test case expect "bigFile.mp4" into resource folder of test module.
```

Comment:
>bigFile.mp4 [](start = 34, length = 11)

Can you just create the file in the test? Why do you think this needs to be tested with this large of a file?

Reply:
We have use case of large files over cloud hence we added test case for large file. There is test case to verify large file in concurrency (5 users push file to cloud) and major memory behavior. If we prepare the large file in buffer (in memory) instead of reading it via stream (file) then that large file it self acquire all the memory.